### PR TITLE
Add sidebar

### PR DESCRIPTION
**Pull request by bprize15**
resolves https://github.com/oncokb/oncokb/issues/3623

### CODE REVIEW

File: src/main/webapp/app/components/sidebar/GenericSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      <div style={{ marginTop: '2rem', marginLeft: '1rem', display: 'flex' }}>
+        <FaChevronRight cursor={'pointer'} style={{ marginTop: '6px' }} onClick={() => toggleGenericSidebar(false)} />
+        {children}
+      </div>
+    </div>
+  ) : (
+    <div className="generic-sidebar-collapsed">
+      <FaChevronLeft
+        cursor={'pointer'}
+        style={{ position: 'absolute', right: '38', top: '38' }}
```
Comment (zhx828) [inline src/main/webapp/app/components/sidebar/GenericSidebar.tsx:24]: What's this magic 38 number?

File: src/main/webapp/app/components/sidebar/GenericSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { observer } from 'mobx-react';
+import { IRootStore } from 'app/stores';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { FaChevronRight, FaChevronLeft } from 'react-icons/fa';
+
+export interface IGenericSidebarProps extends StoreProps {
+  children: React.ReactNode;
+}
+
+const GenericSidebar = ({ children, showGenericSidebar, toggleGenericSidebar, genericSidebarWidth }: IGenericSidebarProps) => {
```
Comment (zhx828) [inline src/main/webapp/app/components/sidebar/GenericSidebar.tsx:12]: Should we just call is Sidebar? Will it collide with components from other library?

**Review by zhx828 — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/components/sidebar/GenericSidebar.tsx (hunk 1 above)
Comment (bprize15) [reply src/main/webapp/app/components/sidebar/GenericSidebar.tsx:24]: It's just a position I chose that makes the chevron that opens the sidebar look better. It also aligns it with the y position of the chevron that closes the sidebar, so the transition is smooth. Should we make it a constant?

File: src/main/webapp/app/components/sidebar/GenericSidebar.tsx (hunk 2 above)
Comment (bprize15) [reply src/main/webapp/app/components/sidebar/GenericSidebar.tsx:12]: It will collide. Changing it to OncoKBSidebar as discussed yesterday

**Review by zhx828 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**